client/asset/eth/multirpc.go

```diff
@@ -840,22 +840,22 @@ func (m *multiRPCClient) withPreferred(f func(*provider) error, acceptabilityFil
 // nonceProviderList returns the randomized provider list, but with any recent
 // nonce provider inserted in the first position.
 func (m *multiRPCClient) nonceProviderList() []*provider {
-	m.providerMtx.Lock()
-	defer m.providerMtx.Unlock()
-
-	providers := make([]*provider, 0, len(m.providers))
-
 	var lastProvider *provider
+	m.lastProvider.Lock()
 	if time.Since(m.lastProvider.stamp) < nonceProviderStickiness {
 		lastProvider = m.lastProvider.provider
 	}
+	m.lastProvider.Unlock()
 
+	m.providerMtx.Lock()
```

Comment:
Both of: mutex currently used by `lastProvider` and `providerMtx` could also be a `sync.RWMutex` (instead of `sync.Mutex`) so we can just lock them for **reads** here (instead of **read-writes**),

not sure whether it will be of any benefit here (probably hard to tell without actually measuring), just wondering what you think of it and whether there are any conventions you follow regarding this (like **if there is read-writes prefer `sync.RWMutex`**) ?

Reply:
> wondering what you think of it and whether there are any conventions you follow regarding this (like **if there is read-writes prefer `sync.RWMutex`**) ?

rule of thumb: if there are frequent reads, and infrequent writes, a RWMutex is sensible.
If reads and writes are similar in occurrence (or mostly write), then plain Mutex

Reply:
There seems to be quite a bit of places where this code is called from, so I'd probably go for `sync.RWMutex` here, 

and `providerMtx` could benefit from `RLock` (instead of `Lock`) in other places as well, so I'd add a commit to change it in those places also if there are no objections

Reply:
My read is that `m.providers` is rarely modified, but accessed very frequently via e.g. `providerList`.  IMO, `sync.RWMutex` fits the bill.  Not the case for `lastProvider` though.

Reply:
changed to `sync.RWMutex` for `providerMtx`: https://github.com/decred/dcrdex/pull/2092/commits/59900ced01a795a2edff0360393724b5ec5ef2a4